addons/vehicleDamage/functions/fnc_processHit.sqf

```diff
@@ -0,0 +1,319 @@
+private _minDamage = [_hitpointConfig >> "minimalHit", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+
+if (_warheadType isEqualTo WARHEAD_TYPE_AP) then {
+    // change damage based on projectile speed (doesn't do this in vanilla ARMA believe it or not)
+    if !(isNull _injurer) then {
+        private _airFriction = [_projectileConfig >> "airFriction", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+        private _distance = _injurer distance _vehicle;
+        _newDamage = (1 - _projectileExplosive) * _newDamage * exp(_airFriction * _distance);
+    };
+};
+
+if (_newDamage < _minDamage) exitWith {
+    TRACE_4("minimum damage returning",_newDamage,_minDamage,_warheadTypeStr,_hitArea);
+    _return
+};
+
+if (_minDamage isEqualTo 0) then {
+    _minDamage = 1;
+};
+
+private _ammoEffectiveness = if (_warheadType isEqualTo WARHEAD_TYPE_AP) then {
+    0.15 max (_newDamage / 500)
+} else {
+    ((_newDamage / _minDamage) * 0.4) min 1
+};
+_incendiary = _incendiary * _ammoEffectiveness;
+
+
+private _isCar = (_vehicle isKindOf "Car" && { !(_vehicle isKindOf "Wheeled_APC_F") });
+if (_isCar) then {
+    _ammoEffectiveness = (_ammoEffectiveness + (_ammoEffectiveness * 0.5)) min 1;
+};
+
+private _injuryChance = 0;
+private _injuryCount = 0;
+switch (_warheadType) do {
+    case WARHEAD_TYPE_AP: {
+        _injuryChance = (_ammoEffectiveness * 2) min 1;
+        _injuryCount = 1 + (_ammoEffectiveness * round random 9);
+    };
+    case WARHEAD_TYPE_HE: {
+        _injuryChance = 0.1; // spalling injury chance alongside direct hit potential
+        _injuryCount = 2;
+        if (_isCar) then {
+            _injuryChance = 0.8;
+            _injuryCount = 3 max random count crew _vehicle;
+        };
+    };
+    default {
+        _injuryChance = (4 * _ammoEffectiveness) min 1;
+        _injuryCount = 2 + round random 3;
+    };
+};
+
+private _currentVehicleAmmo = [_vehicle] call EFUNC(cookoff,getVehicleAmmo);
+private _chanceOfDetonation = 0;
+private _countOfExplodableAmmo = 0;
+if (count (_currentVehicleAmmo select 0) > 0) then {
+    private _magConfig = configFile >> "CfgMagazines";
+    {
+        _x params ["_ammoClassname", "_currentAmmoCount"];
+        private _initialAmmoCount = getNumber(_magConfig >> _ammoClassname >> "count");
+        _chanceOfDetonation = _chanceOfDetonation + (_currentAmmoCount / _initialAmmoCount);
+        _countOfExplodableAmmo = _countOfExplodableAmmo + 1;
+    } forEach (_currentVehicleAmmo select 0);
+    _chanceOfDetonation = _chanceOfDetonation / _countOfExplodableAmmo;
+};
+private _chanceOfDetonate = 0;
+private _chanceOfFire = 0;
+private _currentFuel = fuel _vehicle;
+private _vehicleConfig = _vehicle call CBA_fnc_getObjectConfig;
+switch (_hitArea) do {
+    case "engine": {
+        _chanceOfDetonate = ([_vehicleConfig >> QGVAR(engineDetonationProb), "NUMBER", 0] call CBA_fnc_getConfigEntry) * _incendiary * _currentFuel;
```

Comment:
maybe `_chanceOfDetonation` or `_chanceToDetonate` but chance of detonate is not english